fdbserver/RestoreApplier.actor.cpp

```diff
@@ -88,27 +88,22 @@ ACTOR Future<Void> restoreApplierCore(RestoreApplierInterface applierInterf, int
 }
 
 // The actor may be invovked multiple times and executed async.
-// No race condition as long as we do not wait or yield when operate the shared data, it should be fine,
-// because all actors run on 1 thread.
+// No race condition as long as we do not wait or yield when operate the shared data.
+// Multiple such actors can run on different fileIDs, because mutations in different files belong to different versions;
+// Only one actor can process mutations from the same file
 ACTOR static Future<Void> handleSendMutationVectorRequest(RestoreSendMutationVectorVersionedRequest req,
                                                           Reference<RestoreApplierData> self) {
-	state int numMutations = 0;
+	state NotifiedVersion& curFilePos = self->processedFileState[req.fileIndex];
```

Comment:
To verify if the reference or iterator is pointing to the correct memory location when the `processedFileState ` tree is altered many times.

Reply:
insert won’t invalidate iterator or reference, while erase will invalidate the one to the erased item. The question is answered at https://stackoverflow.com/questions/47280380/can-a-reference-to-an-element-inside-an-stdmap-be-invalidated